warehouse/ingest-core/src/main/java/datawave/iterators/FrequencyColumnIterator.java

```diff
@@ -0,0 +1,179 @@
+import datawave.query.util.Frequency;
+import datawave.query.util.FrequencyFamilyCounter;
+import datawave.query.util.MetadataHelper;
+import datawave.query.util.YearMonthDay;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.PartialKey;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.iterators.user.TransformingIterator;
+import org.apache.hadoop.io.Text;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.HashMap;;
+import java.util.Map;
+
+public class FrequencyColumnIterator extends TransformingIterator {
+    
+    private FrequencyFamilyCounter frequencyFamilyCounter;
+    // TODO Figure out how to keep the rowIdToCompressedFreqCQMap from getting too large
+    // TODO maybe we do not have to worry about that from happening.
+    private HashMap<String,FrequencyFamilyCounter> rowIdToCompressedFreqCQMap = new HashMap<>();
+    private String ageOffDate = "19000101";
+    
+    public FrequencyColumnIterator() {}
+    
+    public FrequencyColumnIterator(FrequencyColumnIterator aThis, IteratorEnvironment environment) {
+        super();
+        setSource(aThis.getSource().deepCopy(environment));
+    }
+    
+    @Override
+    public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options, IteratorEnvironment env) throws IOException {
+        super.init(source, options, env);
+        if (options.get("ageOffDate") != null)
+            ageOffDate = options.get("ageOffDate");
+    }
+    
+    @Override
+    protected PartialKey getKeyPrefix() {
+        return PartialKey.ROW_COLFAM;
+    }
+    
+    @Override
+    protected void transformRange(SortedKeyValueIterator<Key,Value> sortedKeyValueIterator, KVBuffer kvBuffer) throws IOException {
+        Key newKey = null;
+        Long numRecords = 0L;
+        frequencyFamilyCounter = new FrequencyFamilyCounter();
+        
+        if (log.isTraceEnabled())
+            log.trace("Transforming range for key " + sortedKeyValueIterator.getTopKey().getRow().toString(), new Exception());
+        
+        Key topKey = null, aggregatedKey = null;
+        Value topValue = null;
+        Value aggregatedValue = null;
+        boolean isFrequencyRange = false;
+        
+        while (sortedKeyValueIterator.hasTop()) {
+            numRecords++;
+            Text cq = sortedKeyValueIterator.getTopKey().getColumnQualifier();
+            topKey = sortedKeyValueIterator.getTopKey();
+            topValue = sortedKeyValueIterator.getTopValue();
+            
+            // Just do an identity transform if there it is not a Frequency Column
+            if (!topKey.getColumnFamily().toString().equals(ColumnFamilyConstants.COLF_F.toString())) {
+                kvBuffer.append(sortedKeyValueIterator.getTopKey(), sortedKeyValueIterator.getTopValue());
```

Comment:
Oh SNAP.  I just figured out something critical here.  We need a separate frequency family column PER datatype!!!!   That is that part of the column qualifier that is before the date.  Also it appears that the last character of the datatype is being trimmed off somehow.